annotated.go

```diff
@@ -195,6 +196,310 @@ func ResultTags(tags ...string) Annotation {
 	return resultTagsAnnotation{tags}
 }
 
+type _lifecycleHookAnnotationType int
+
+const (
+	_unknownHookType _lifecycleHookAnnotationType = iota
+	_onStartHookType
+	_onStopHookType
+)
+
+type lifecycleHookAnnotation struct {
+	Type   _lifecycleHookAnnotationType
+	Target interface{}
+}
+
+func (la *lifecycleHookAnnotation) apply(ann *annotated) error {
+	name := "UnknownHookAnnotation"
+	switch la.Type {
+	case _onStartHookType:
+		name = _onStartHook
+	case _onStopHookType:
+		name = _onStopHook
+	}
```

Comment:
Looks like name is only used in errors here, can we save an alloc for successful cases by implementing a stringer interface for `_lifecycleHookAnnotationType` and moving this function there.